examples/js/modifiers/TessellateModifier.js

```diff
@@ -2,273 +2,216 @@
 
 	this.maxEdgeLength = maxEdgeLength;
 	this.maxIterations = maxIterations;
-	this.maxFaces = maxFaces;
 
 };
 
-// Applies the "modify" pattern
 THREE.TessellateModifier.prototype.modify = function ( geometry ) {
 
-	const isBufferGeometry = geometry.isBufferGeometry;
+	if ( geometry.isBufferGeometry !== true ) {
 
-	if ( isBufferGeometry ) {
-
-		geometry = new THREE.Geometry().fromBufferGeometry( geometry );
-
-	} else {
-
-		geometry = geometry.clone();
+		console.warn( 'TessellateModifier: geometry is not a BufferGeometry.', geometry );
+		return geometry;
 
 	}
 
-	geometry.mergeVertices( 6 );
+	//
 
-	let finalized = false;
-	let iteration = 0;
+	const maxIterations = this.maxIterations;
 	const maxEdgeLengthSquared = this.maxEdgeLength * this.maxEdgeLength;
 
-	let edge;
-
-	while ( ! finalized && iteration < this.maxIterations && geometry.faces.length < this.maxFaces ) {
-
-		const faces = [];
-		const faceVertexUvs = [];
-
-		finalized = true;
-		iteration ++;
-
-		for ( var i = 0, il = geometry.faceVertexUvs.length; i < il; i ++ ) {
-
-			faceVertexUvs[ i ] = [];
-
-		}
-
-		for ( var i = 0, il = geometry.faces.length; i < il; i ++ ) {
-
-			const face = geometry.faces[ i ];
-
-			if ( face instanceof THREE.Face3 ) {
-
-				const a = face.a;
-				const b = face.b;
-				const c = face.c;
-
-				const va = geometry.vertices[ a ];
-				const vb = geometry.vertices[ b ];
-				const vc = geometry.vertices[ c ];
-
-				const dab = va.distanceToSquared( vb );
-				const dbc = vb.distanceToSquared( vc );
-				const dac = va.distanceToSquared( vc );
-
-				const limitReached = ( faces.length + il - i ) >= this.maxFaces;
-
-				if ( ! limitReached && ( dab > maxEdgeLengthSquared || dbc > maxEdgeLengthSquared || dac > maxEdgeLengthSquared ) ) {
-
-					finalized = false;
-
-					const m = geometry.vertices.length;
-
-					const triA = face.clone();
-					const triB = face.clone();
-
-					if ( dab >= dbc && dab >= dac ) {
-
-						var vm = va.clone();
-						vm.lerp( vb, 0.5 );
-
-						triA.a = a;
-						triA.b = m;
-						triA.c = c;
-
-						triB.a = m;
-						triB.b = b;
-						triB.c = c;
+	const va = new THREE.Vector3();
+	const vb = new THREE.Vector3();
+	const vc = new THREE.Vector3();
+	const vm = new THREE.Vector3();
+	const vs = [ va, vb, vc, vm ];
 
-						if ( face.vertexNormals.length === 3 ) {
+	const na = new THREE.Vector3();
+	const nb = new THREE.Vector3();
+	const nc = new THREE.Vector3();
+	const nm = new THREE.Vector3();
+	const ns = [ na, nb, nc, nm ];
 
-							var vnm = face.vertexNormals[ 0 ].clone();
-							vnm.lerp( face.vertexNormals[ 1 ], 0.5 );
+	const ca = new THREE.Color();
+	const cb = new THREE.Color();
+	const cc = new THREE.Color();
+	const cm = new THREE.Color();
+	const cs = [ ca, cb, cc, cm ];
 
-							triA.vertexNormals[ 1 ].copy( vnm );
-							triB.vertexNormals[ 0 ].copy( vnm );
+	/*TOFIX?*/
 
-						}
+	THREE.Color.prototype.lerpColors = function lerpColors( color1, color2, alpha ) {
```

Comment:
We should probably just add this to `THREE.Color`? 🤔

Reply:
There is also `Vector3.lerpVectors()`. So `Color.lerpColors()` makes totally sense 👍 .

Reply:
Done! #21061